drivers/kscan/kscan_gt911.c

```diff
@@ -96,22 +96,29 @@ static int gt911_process(const struct device *dev)
 		return 0;
 	}
 
-	/* need to clear the status */
-	uint8_t clear_buffer[3] = {(uint8_t)REG_STATUS, (uint8_t)(REG_STATUS >> 8), 0};
-
-	r = i2c_write_dt(&config->bus, clear_buffer, sizeof(clear_buffer));
-	if (r < 0) {
-		return r;
-	}
-
-	/* obtain first point X, Y coordinates and event from:
-	 * REG_P1_XH, REG_P1_XL, REG_P1_YH, REG_P1_YL.
-	 */
-	reg_addr = REG_FIRST_POINT;
-	r = i2c_write_read_dt(&config->bus, &reg_addr, sizeof(reg_addr),
-							&pointRegs, sizeof(pointRegs));
-	if (r < 0) {
-		return r;
+	if (status & TOUCH_STATUS_MSK) {
```

Comment:
Could you invert the condition and keep executing inline instead? Like

```
if (!(status & TOUCH_STATUS_MSK)) {
		/* Status bit not set, ignore this event */
		return 0;
}
...
```

Reply:
Sure, this is much cleaner. Updated PR with this change